ansible/undercloud/deploy-undercloud.yml

```diff
@@ -215,16 +215,43 @@
         owner: stack
         group: stack
 
-    - name: check if credentials in instackenv.json are correct
+    - name: Newton(10)/Ocata(11)/Pike(12) check if credentials in instackenv.json are correct
       shell: openstack baremetal instackenv validate  -f  /home/stack/instackenv.json
       register: node_status
       failed_when: "'SUCCESS' not in node_status.stdout"
-
-    - name: Import instackenv.json
-      shell: . /home/stack/stackrc; openstack baremetal import --json /home/stack/instackenv.json
-
-    - name: Configure boot
-      shell: . /home/stack/stackrc; openstack baremetal configure boot
+      when: version < 13
+
+    - name: Queens(13) Import instackenv.json
+      shell: |
+        . /home/stack/stackrc
+        openstack overcloud node import /home/stack/instackenv.json
+      when: version >= 13
+
+    - name: Newton(10)/Ocata(11)/Pike(12) Import instackenv.json
+      shell: |
+        . /home/stack/stackrc
+        openstack baremetal import --json /home/stack/instackenv.json
+      when: version < 13
+
+    - name: Queens(13) Configure boot
+      shell: |
+        . /home/stack/stackrc
+        openstack overcloud node configure --all-manageable
+      when: version >= 13
+
+    - name: Newton(10)/Ocata(11)/Pike(12) Configure boot
+      shell: |
+        . /home/stack/stackrc
+        openstack baremetal configure boot
+      when: version < 13
+
+    - name: Queens(13) Set Nodes to provide
+      shell: |
```

Comment:
Why are we doing this before introspection? Shouldn't nodes be in manageable for introspection?

Reply:
Also, we do not need to set the nodes to manageable explicitly since it will be done in introspect.sh